Libraries/PushNotificationIOS/PushNotificationIOS.js

```diff
@@ -23,6 +23,15 @@ var DEVICE_NOTIF_EVENT = 'remoteNotificationReceived';
 var NOTIF_REGISTER_EVENT = 'remoteNotificationsRegistered';
 var DEVICE_LOCAL_NOTIF_EVENT = 'localNotificationReceived';
 
+const REPEAT_INTERVAL = {
+   year: 4,
+   month: 8,
+   week: 8192,
+   day: 16,
+   hour: 32,
+   minute: 64
+};
```

Comment:
What if we just remove that mapping, pass the `string` value directly to Javascript (no need to check the `details` on JS side) and define something like this (note - you have to check proper enum names):
```objc
@implementation RCTConvert (NSCalendarUnit)

RCT_ENUM_CONVERTER(NSCalendarUnit, (@{
  @"era": @(NSCalendarUnitEra),
  @"week": @(NSCalendarUnitWeekday)
}), 0, integerValue)

@end
```

That means we can just:
```objc
notification.repeatInterval = [RCTConvert NSCalendarUnit:details[@"repeatInterval"]];
```
- default value of `0` will be given for null, undefined, and non-matching strings